### PR TITLE
refactor: Split consumer.rs file

### PR DESCRIPTION
**Pull request by CleverAkanoa**
This PR aims to ease the code navigability of the consumer part.

consumer.rs involves more than 2000 lines of codes, it's quite hard to know where you are.

Even all structure are related to consumer, they can be split in more than one files.

No code has been changed, only moved and visibility opened when it was mandatory.

### CODE REVIEW

**Review by FlorentinDUBOIS — APPROVED** [review]

**Comment by KannarFr** [comment]
So much easier to read! <3.

**Comment by tisonkun** [comment]
@CleverAkanoa is this PR ready for review now?

**Comment by CleverAkanoa** [comment]
I think so :)

File: src/consumer/message.rs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+///
+/// it is generic over the type it can be deserialized to
+pub struct Message<T> {
+    /// origin topic of the message
+    pub topic: String,
+    /// contains the message's data and other metadata
+    pub payload: Payload,
+    /// contains the message's id and batch size data
+    pub message_id: MessageData,
+    pub(crate) _phantom: PhantomData<T>,
```
Comment (tisonkun) [inline src/consumer/message.rs]:
```suggestion
    pub(super) _phantom: PhantomData<T>,
```

Narrow the scope as possible. Previously it was private.

File: src/consumer/topic.rs [hunk 2]
```diff
@@ -0,0 +1,374 @@
...
+    pub(crate) consumer_id: u64,
+    pub(crate) config: ConsumerConfig,
+    topic: String,
+    messages: Pin<Box<MessageIdDataReceiver>>,
+    engine_tx: mpsc::UnboundedSender<EngineMessage<Exe>>,
+    #[allow(unused)]
+    data_type: PhantomData<fn(Payload) -> T::Output>,
+    pub(crate) dead_letter_policy: Option<DeadLetterPolicy>,
+    pub(crate) last_message_received: Option<DateTime<Utc>>,
+    pub(crate) messages_received: u64,
```
Comment (tisonkun) [inline src/consumer/topic.rs]:
```suggestion
    pub(super) last_message_received: Option<DateTime<Utc>>,
    pub(super) messages_received: u64,
```

ditto

File: src/consumer/topic.rs [hunk 3]
```diff
@@ -0,0 +1,374 @@
...
+}
+
+impl<T: DeserializeMessage, Exe: Executor> TopicConsumer<T, Exe> {
+    #[cfg_attr(feature = "telemetry", tracing::instrument(skip_all))]
+    pub async fn new(
```
Comment (tisonkun) [inline src/consumer/topic.rs]:
```suggestion
    pub(super) async fn new(
```

**Review by tisonkun — COMMENTED** [review]
Thanks for your effort! Minor comments inline.

**Review by tisonkun — APPROVED** [review]
Thanks for your contribution @CleverAkanoa!

Merging...

**Comment by CleverAkanoa** [comment]
Thanks @tisonkun <3

File: src/consumer/multi.rs [hunk 4]
```diff
@@ -0,0 +1,338 @@
+use std::{
+    collections::{BTreeMap, VecDeque},
+    fmt::Debug,
+    future::Future,
+    iter,
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+use async_std::prelude::Stream;
```
Comment (lukesteensen) [inline src/consumer/multi.rs:10]: This causes a compile error when the `async_std` feature is not active.

File: src/consumer/topic.rs [hunk 5]
```diff
@@ -0,0 +1,374 @@
+use std::{
+    marker::PhantomData,
+    pin::Pin,
+    sync::Arc,
+    task::{Context, Poll},
+    time::{Duration, Instant},
+};
+
+use async_std::prelude::Stream;
```
Comment (lukesteensen) [inline src/consumer/topic.rs:9]: Same here.

**Review by lukesteensen — COMMENTED** [review]
Hello! I just noticed some errors when we bumped from 5.0.0 to 5.0.1 and tracked them back to this PR. Both due to using `async_std` when we don't have that feature active.

File: src/consumer/multi.rs (hunk 4 above)
Comment (tisonkun) [reply src/consumer/multi.rs:10]: We can use `futures::Stream` which is runtime agnostic. I filed a patch: https://github.com/streamnative/pulsar-rs/pull/250.

**Pull request merged**